### PR TITLE
added stride parameter to sample_first_frames

### PR DESCRIPTION
**Pull request by yashbhalgat**
Possibly less effective, but much faster than `sliding_frames_sampling`.

### CODE REVIEW

File: eta/core/video.py [hunk 1]
```diff
@@ -1530,7 +1530,7 @@ def sample_select_frames(video_path, frames, output_patt=None, fast=False):
         return imgs
 
 
-def sample_first_frames(arg, k, size=None):
+def sample_first_frames(arg, k, stride=2, size=None):
```
Comment (brimoor) [inline eta/core/video.py]: you're changing the default behavior! should be `stride=1`

Comment (brimoor) [reply eta/core/video.py]: need to document new parameter in docstring

File: eta/core/video.py [hunk 2]
```diff
@@ -1546,11 +1546,12 @@ def sample_first_frames(arg, k, size=None):
     # Read frames ...
     if isinstance(arg, six.string_types):
         # ... from disk
-        with FFmpegVideoReader(arg, frames="1-%d" % k) as vr:
+        with FFmpegVideoReader(
```
Comment (brimoor) [inline eta/core/video.py]: needs to be `range(1, stride * k  + 1, stride)`

File: eta/core/video.py [hunk 3]
```diff
@@ -1546,11 +1546,12 @@ def sample_first_frames(arg, k, size=None):
...
+            arg, frames=[i for i in range(1, stride*k, stride)]) as vr:
             imgs = [img for img in vr]
     else:
         # ... from tensor
-        imgs = arg[:k]
+        imgs = arg[:k*stride:stride]
```
Comment (brimoor) [inline eta/core/video.py]: style: one space before and after `*`. Optional `()` for clarity: `(k * stride)`

File: eta/core/video.py (hunk 1 above)
Comment (yashbhalgat) [reply eta/core/video.py]: Oh right, my bad.

**Pull request merged**